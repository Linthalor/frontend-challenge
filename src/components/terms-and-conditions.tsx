import { Box, Button, Card, CardActions, CardContent, Modal, Typography } from '@mui/material';
import { forwardRef, Ref, useImperativeHandle, useState } from 'react';
import terms from '../terms';

export interface TermsAndConditionsRef {
  setOpenTerms: React.Dispatch<React.SetStateAction<boolean>>
}

export const TermsAndConditions = forwardRef((_: {}, ref: Ref<TermsAndConditionsRef>) => {
  const [openTerms, setOpenTerms] = useState(false);
  useImperativeHandle(ref, () => ({ setOpenTerms }));
  return (
    <Modal
      open={openTerms}
      onClose={() => setOpenTerms(false)}
      aria-labelledby="modal-modal-title"
      aria-describedby="modal-modal-description"
    >
      <Card
        // TODO: use media queries to make this go full screen on small devices
        sx={{
          position: 'absolute',
          top: '50%',
          left: '50%',
          transform: 'translate(-50%, -50%)',
          width: 600,
          height: 800,
          maxHeight: '80vh',
          display: 'flex',
          flexDirection: 'column'
        }}
      >
        <CardContent
          sx={{
            overflow: 'hidden',
            display: 'flex',
            flexDirection: 'column',
            flex: 1
          }}
        >
          <Typography id="modal-modal-title" variant="h6" component="h2">
            Terms & Conditions
          </Typography>
          <Box sx={{ mt: 2, overflowY: 'auto', flex: 1 }}>
            <Typography id="modal-modal-description" sx={{ whiteSpace: 'pre-wrap' }}>
              {terms}
            </Typography>
          </Box>
        </CardContent>
        <CardActions
          sx={{
            display: 'flex',
            flexDirection: 'row',
            justifyContent: 'end',
          }}
        >
          <Button onClick={() => setOpenTerms(false)}>Close</Button>
        </CardActions>
      </Card>
    </Modal>
  );
});