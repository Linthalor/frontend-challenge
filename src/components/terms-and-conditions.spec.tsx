import { cleanup, render } from '@testing-library/react';
import { TermsAndConditions, TermsAndConditionsRef } from './terms-and-conditions';
import { act } from 'react-dom/test-utils';

jest.mock('@mui/material');

describe('TermsAndConditions', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders as expected closed', () => {
    const { container } = render(
      <TermsAndConditions />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders as expected open', () => {
    let ref: TermsAndConditionsRef | null = null;
    const { container } = render(
      <TermsAndConditions ref={r => ref = r} />
    );

    expect(container.firstChild).toMatchSnapshot();

    act(() => {
      const button = container.querySelector('button');
      ref?.setOpenTerms(true);
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});