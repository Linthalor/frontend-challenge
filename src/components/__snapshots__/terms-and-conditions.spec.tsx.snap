// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsAndConditions renders as expected closed 1`] = `null`;

exports[`TermsAndConditions renders as expected open 1`] = `null`;

exports[`TermsAndConditions renders as expected open 2`] = `null`;
